src/python/pants/backend/python/lint/flake8/subsystem.py

```diff
@@ -34,3 +37,16 @@ def register_options(cls, register):
             advanced=True,
             help="Path to `.flake8` or alternative Flake8 config file",
         )
+        register(
+            "--output-file",
+            type=str,
+            metavar="filename",
+            default=None,
+            advanced=True,
+            help="Redirect report to a file.",
+        )
```

Comment:
Rather than an output file, could this potentially be an output directory, located on the `Lint` `Goal`? That way each linter could emit its output in subdirectories of a shared output directory... and we could probably give it a reasonable default so that reports are always generated.

Since the `LintOutputFile` type is exposed to the `Goal`, it would be good to enable more linters to be able to use it (without specifying an output file per linter).

Reply:
ok, updated.
But keep in mind that now we hard code the report file name for each linter in the rule.
Which is not optimal, since each linter has other options to control the format of the file (using pass-thru args or by adding plugins to the requirements).
so the issue there is that the file name will not reflect the format of the file.
We can obviously expose the file name as an option on each linter, but then, a user will have to configure things in two places, which is confusing.

Reply:
True... but it feels like whether or not a linter has multiple file formats is a per linter concern... the linter can choose which filename it wants to _within_ the directory. Fine either way.